lib/livebook/session.ex

```diff
@@ -72,7 +75,14 @@ defmodule Livebook.Session do
           runtime_monitor_ref: reference() | nil,
           autosave_timer_ref: reference() | nil,
           save_task_pid: pid() | nil,
-          saved_default_file: FileSystem.File.t() | nil
+          saved_default_file: FileSystem.File.t() | nil,
+          system_memory_timer_ref: reference() | nil,
+          memory_usage: memory_usage() | Livebook.Utils.system_memory() | nil
+        }
+
+  @type memory_usage :: %{
+          node: Livebook.Runtime.node_memory(),
```

Comment:
```suggestion
          node: Livebook.Runtime.node_memory() | nil,
```

Reply:
Also, maybe name it `Livebook.Runtime.memory_usage()`. And then use `runtime` instead of `node` for consistency?